src/mgmt/srv_drpc.c

```diff
@@ -90,8 +91,15 @@ ds_mgmt_drpc_kill_rank(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	/* Response status is populated with SUCCESS on init. */
 	mgmt__daos_resp__init(resp);
 
-	/* TODO: do something with request and populate daos response status */
-
+	/* terminate local service */
+	if (req->force)
+		sig = SIGKILL;
+	else
+		sig = SIGTERM;
+	D_INFO("Service rank %d is being killed by signal %d\n",
```

Comment:
Don't we need to look up the specified rank and send a message to that node? Perhaps I am missing something.

Reply:
it's node local at this point

Reply:
will be executing on the rank in the request